modules/jsonld/src/test/scala/ch/epfl/bluebrain/nexus/rdf/jsonld/JsonLdSpec.scala

```diff
@@ -0,0 +1,250 @@
+package ch.epfl.bluebrain.nexus.rdf.jsonld
+
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.rdf.RdfSpec
+import ch.epfl.bluebrain.nexus.rdf.jsonld.JsonLd.{CircularContextDependency, ContextNotFound, IllegalContextValue}
+import ch.epfl.bluebrain.nexus.rdf.jsonld.syntax._
+import ch.epfl.bluebrain.nexus.rdf.syntax.all._
+import io.circe.literal._
+
+import scala.util.{Failure, Success, Try}
+
+class JsonLdSpec extends RdfSpec {
+
+  "JsonLd" should {
+    "resolve contexts" when {
+      "no @context is present" in {
+        val json = json"{}"
+
+        json
+          .resolveContext[Try](_ => Failure(new IllegalArgumentException))
+          .value
+          .success
+          .value
+          .rightValue shouldEqual
+          json"""
+          {
+            "@context": {}
+          }
+          """
+      }
+
+      "context is an IRI" in {
+        val contextUri = url"http://context.example.com"
+
+        val json =
+          json"""
+          {
+            "@context": ${contextUri.asString}
+          }
+          """
+
+        val contextValue =
+          json"""
+          {
+            "@context": {
+              "schema": "http://schema.org/"
+            }
+          }       
+          """
+
+        json
+          .resolveContext[Try] {
+            case `contextUri` => Success(Some(contextValue))
+            case _            => Failure(new IllegalArgumentException)
+          }
+          .value
+          .success
+          .value
+          .rightValue shouldEqual (json deepMerge contextValue)
+      }
+
+      "context is a JSON array containing IRI" in {
+        val contextUri = url"http://context.example.com"
+
+        val json =
+          json"""
+          {
+            "@context": [
+              {
+                "xsd": "http://www.w3.org/2001/XMLSchema#"
+              },
+              ${contextUri.asString}
+            ]
+          }                
+          """
+
+        val contextValue =
+          json"""
+          {
+            "@context": {
+              "schema": "http://schema.org/"
+            }
+          }       
+          """
+
+        val expected =
+          json"""
+          {
+            "@context": {
+              "xsd": "http://www.w3.org/2001/XMLSchema#",
+              "schema": "http://schema.org/"
+            }
+          }               
+          """
+
+        json
+          .resolveContext[Try] {
+            case `contextUri` => Success(Some(contextValue))
+            case _            => Failure(new IllegalArgumentException)
+          }
+          .value
+          .success
+          .value
+          .rightValue shouldEqual expected
+      }
+      "when contexts are nested" in {
```

Comment:
We also need to test scoped contexts. A test for this json would be useful:
```json
{
  "@context": {
    "xsd": "http://www.default-xsd.com#"
  },
  "key": {
    "@context": {
      "xsd": "http://www.w3.org/2001/XMLSchema#"
    },
    "key2": {
      "@context": {
        "schema": "http://schema.org/"
      }
    }
  }
}
```

This would also test the overriding of the key `xsd`